crates/cdk/src/wallet/send.rs

```diff
@@ -45,135 +29,351 @@ impl Wallet {
             }
+        }
+        tracing::debug!(
+            "Proofs to send: {:?}",
+            proofs_to_send.iter().map(|p| p.amount).collect::<Vec<_>>()
+        );
 
-            // Handle online tolerance when selection fails and conditions are present
-            (SendKind::OnlineTolerance(_), Err(_), Some(_)) => {
-                tracing::info!("Could not select proofs with conditions while offline.");
-                tracing::info!("Attempting to select proofs without conditions and swapping");
+        // Check if sufficient proofs are available
+        if send.amount > proofs_to_send.total_amount()? {
+            return Err(Error::InsufficientFunds);
+        }
 
-                self.swap_from_unspent(amount, conditions, include_fees)
-                    .await?
-            }
+        // Update proofs state to pending spent
+        self.localstore
+            .update_proofs_state(proofs_to_send.ys()?, State::PendingSpent)
+            .await?;
 
-            // Handle online tolerance with successful selection
-            (SendKind::OnlineTolerance(tolerance), Ok(selected_proofs), _) => {
-                let selected_proofs_amount = selected_proofs.total_amount()?;
-                let amount_to_send = match include_fees {
-                    true => amount + self.get_proofs_fee(&selected_proofs).await?,
-                    false => amount,
-                };
-                if selected_proofs_amount - amount_to_send <= *tolerance {
-                    selected_proofs
-                } else {
-                    tracing::info!("Could not select proofs while offline. Attempting swap");
-                    self.swap_from_unspent(amount, conditions, include_fees)
-                        .await?
-                }
-            }
+        // Include token memo
+        let send_memo = send.options.memo.or(memo);
+        let memo = send_memo.and_then(|m| if m.include_memo { Some(m.memo) } else { None });
 
-            // Handle all other cases where selection fails
-            (
-                SendKind::OfflineExact
-                | SendKind::OnlineExact
-                | SendKind::OfflineTolerance(_)
-                | SendKind::OnlineTolerance(_),
-                Err(_),
-                _,
-            ) => {
-                tracing::debug!("Could not select proofs");
-                return Err(Error::InsufficientFunds);
-            }
-        };
+        // Create and return token
+        Ok(Token::new(
+            self.mint_url.clone(),
+            proofs_to_send,
+            memo,
+            self.unit.clone(),
+        ))
+    }
+
+    /// Cancel prepared send
+    pub async fn cancel_send(&self, send: PreparedSend) -> Result<(), Error> {
+        tracing::info!("Cancelling prepared send");
+
+        self.localstore
+            .update_proofs_state(send.proofs().ys()?, State::Unspent)
+            .await?;
+
+        Ok(())
+    }
+}
+
+/// Prepared send
```

Comment:
Can we just add a few more lines explaining how the `proofs_to_swap` vs `proofs_to_send` should be uses and what they are?